src/sentry/tasks/integrations/sync_status_inbound.py

```diff
@@ -51,17 +57,78 @@ def sync_status_inbound(
         "provider_key": provider.key,
         "integration_id": integration_id,
     }
-
     if action == ResolveSyncAction.RESOLVE:
+        activity_type = ActivityType.SET_RESOLVED
+        # If the resolution strategy is set to resolve in the next release or current release
+        if config.get("resolution_strategy") in [
+            "resolve_next_release",
+            "resolve_current_release",
+        ]:
+            # only allow setting resolved in release if there is a release for each project
+            all_project_ids = list({group.project_id for group in affected_groups})
+            has_releases_for_each_project = all(
+                Release.objects.filter(
+                    projects=project_id, organization_id=organization_id, status=ReleaseStatus.OPEN
+                ).exists()
+                for project_id in all_project_ids
+            )
+            if has_releases_for_each_project:
+                if config.get("resolution_strategy") == "resolve_next_release":
+                    activity_type = ActivityType.SET_RESOLVED_IN_RELEASE
+                    activity_data["inNextRelease"] = True
+                elif config.get("resolution_strategy") == "resolve_current_release":
+                    activity_type = ActivityType.SET_RESOLVED_IN_RELEASE
+
         Group.objects.update_group_status(
             groups=affected_groups,
             status=GroupStatus.RESOLVED,
             substatus=None,
-            activity_type=ActivityType.SET_RESOLVED,
+            activity_type=activity_type,
             activity_data=activity_data,
         )
-
         for group in affected_groups:
+            # update the resolutions
+            # probably should be done within a single transaction with the status but this is fine for now
+            # note this logic is ported from src/sentry/api/helpers/group_index/update.py
+            if activity_type == ActivityType.SET_RESOLVED_IN_RELEASE:
+                # find the latest release by date for the project
+                last_release_by_date = (
+                    Release.objects.filter(
+                        projects=group.project,
+                        organization_id=organization_id,
+                        status=ReleaseStatus.OPEN,
+                    )
+                    .extra(select={"sort": "COALESCE(date_released, date_added)"})
+                    .order_by("-sort")
+                    .first()
+                )
+                # Check if semver versioning scheme is followed
+                follows_semver = follows_semver_versioning_scheme(
+                    org_id=organization_id,
+                    project_id=group.project.id,
+                    release_version=last_release_by_date.version,
+                )
+
+                resolution_params = {
+                    "status": GroupStatus.RESOLVED,
+                    "release": last_release_by_date,  # Is this the right release?
```

Comment:
for `resolve_next_release` should we be trying to get the next release? i think we do this in `update.py` if `follows_semver` is `False`